app/models/certificate.rb

```diff
@@ -52,43 +52,52 @@ def self_signed?
     signing_key_id == key_id
   end
 
-  def validate_cert
+  def validate_cert(is_leaf: false)
     if expired?
       'expired'
     elsif trusted_root?
       # The other checks are all irrelevant if we trust the root.
+      raise 'trusted root missing from store' if CertificateStore.instance[key_id].blank?
```

Comment:
should we add the `key_id` to the error message? or the subject of the cert?

Reply:
added key_id